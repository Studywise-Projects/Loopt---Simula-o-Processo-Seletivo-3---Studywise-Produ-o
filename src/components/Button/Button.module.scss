@import '@/styles/_global.scss';

.button {
  box-sizing: border-box;
  flex-shrink: 0;
  height: 44px;
  padding: 12px 14px;
}

.auxButton {
  width: min-content;
}

.contained {
  background-color: $primary-color;
  color: $white;
}

.text {
  background-color: none;
  color: $primary-color;
  max-width: min-content;
}

.outlined {
  background-color: none;
  color: $primary-color;
  border: 1px solid $primary-color;
}