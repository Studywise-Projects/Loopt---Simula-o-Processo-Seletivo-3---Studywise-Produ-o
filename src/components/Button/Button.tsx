import { Button as ButtonMui } from '@mui/material';
import LoadingButton from '@mui/lab/LoadingButton';
import styles from './Button.module.scss';
import classNames from 'classnames';

export interface IButton {
  variant?: 'contained' | 'text' | 'outlined';
  isAuxButton?: boolean;
  isSecondaryButton?: boolean;
  isLoading?: boolean;
  text: string;
  handleClick: () => void;
}

function Button({
  variant = 'contained',
  isAuxButton = false,
  isSecondaryButton = false,
  isLoading = false,
  text,
  handleClick,
}: IButton) {
  const buttonStyles = classNames(styles.button, {
    [styles[variant]]: variant,
    [styles.auxButton]: isAuxButton === true,
    [styles.secondaryButton]: isSecondaryButton === true,
  });

  return (
    <>
      {isLoading === false ? (
        <ButtonMui
          data-test-id='button'
          variant={variant}
          onClick={handleClick}
          className={buttonStyles}
        >
          {text}
        </ButtonMui>
      ) : (
        <LoadingButton
          loading
          className={buttonStyles}
          variant='contained'
        ></LoadingButton>
      )}
    </>
  );
}

export default Button;
