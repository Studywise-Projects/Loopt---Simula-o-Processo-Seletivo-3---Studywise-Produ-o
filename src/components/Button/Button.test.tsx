import { render, fireEvent, getByTestId } from '@testing-library/react';
import Button from './Button';

describe('Button', () => {
  it('Should match the snapshot', () => {
    const { getByText } = render(
      <Button text='Button' handleClick={undefined} />,
    );

    const element = getByText('Button');
    expect(element).toMatchSnapshot();
  });

  it('Should render correctly', () => {
    const { getByText } = render(
      <Button text='Button' handleClick={undefined} />,
    );

    const element = getByText('Button');
    expect(element).toBeInTheDocument();
  });

  it('Should call handleClickMock function once', () => {
    const handleClickMock = jest.fn();

    const { getByText } = render(
      <Button text='Button' handleClick={handleClickMock} />,
    );

    const element = getByText('Button');
    fireEvent.click(element);

    expect(handleClickMock).toHaveBeenCalledTimes(1);
  });
});
