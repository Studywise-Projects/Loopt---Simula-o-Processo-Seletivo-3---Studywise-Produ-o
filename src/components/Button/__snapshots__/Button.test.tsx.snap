// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Should match the snapshot 1`] = `
<button
  class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-sghohy-MuiButtonBase-root-MuiButton-root"
  data-test-id="button"
  tabindex="0"
  type="button"
>
  Button
  <span
    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
  />
</button>
`;
